test/compare_results.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+from pytest import approx
+
+QUANTUM_ENERGY_IDX=4
+
+def find_start_end_block(lines, blockname):
+    """Identifies start and end of blocks"""
+    for i, line in enumerate(lines):
+        if "Start " + blockname in line:
+            start_idx = i
+        elif "End " + blockname in line:
+            end_idx = i
+    return start_idx, end_idx
+
+def extract_quantum_block(lines):
+    """Extracts only the quantum block from a chk file"""
+    idxs = find_start_end_block(lines, "Quantum")
+    temp = lines[idxs[0]+4:idxs[-1]]
+    return [line.split() for line in temp]
+
+def extract_quantum_energies(block):
+    return [float(line[QUANTUM_ENERGY_IDX]) for line in block]
+
+def read_chk_file(fname):
+    with open(fname, 'r') as fp:
+        lines = fp.readlines()
+        if lines:
+            lines  = filter( lambda x : x != '\n', lines ) # remove empty lines
+            lines  = filter( lambda x : '<-' not in x, lines ) # remove commented lines
+    return list(lines)
+
+def read_energy_column(fname, column_no):
+    lines = read_chk_file(fname)
+    lines = lines[:-1] # remove last line
+    return [float(line.split()[column_no]) for line in lines]
+
+def read_quantum_block(fname):
+    lines = read_chk_file(fname)
+    return extract_quantum_block(lines)
+
+def compare_columns(fname1, fname2, column_no, precision=1e-10):
+    energies1 = read_energy_column(fname1, column_no)
+    energies2 = read_energy_column(fname2, column_no)
+
+    for e1, e2 in zip(energies1, energies2):
+        if not e1 == approx(e2, abs=precision):
+            print(e1, e2, abs(e1-e2))
+            raise AssertionError
```

Comment:
I think it's more natural to have the error message as part of the assertion:
```suggestion
        assert e1 == approx(e2, abs=precision), \
            f"{e1} and {e2} differ by {abs(e1-e2)}"
```

Reply:
That said, since you're using `pytest` anyway, why not write test functions in that format and invoke `pytest`, instead of running this as a script?

Reply:
I wasn't initially sure if pytest could handle command line arguments (since we're using this script as a CLI tool) but it looks like it can. Is this too complex a use-case for future maintainers though?